pandas/core/frame.py

```diff
@@ -7898,6 +7915,19 @@ def _get_data(axis_matters):
                 raise NotImplementedError(msg)
             return data
 
+        def blk_func(values):
+            if isinstance(values, ExtensionArray):
```

Comment:
with this inside blk_func, shouldn't the block-wise operation have the same performance bump as the column-wise?

Reply:
> with this inside blk_func, shouldn't the block-wise operation have the same performance bump as the column-wise?

It didn't actually change anything performance wise (it's the same function being called as before). 
The reason that both paths have different performance, is because the re-assembling of the results into a Series is more expensive for the block-wise compared to column-wise. 

(it's possible that the block-wise way could be optimized to get rid of this difference though. The main thing is that the block results are not in order)

Reply:
> (it's possible that the block-wise way could be optimized to get rid of this difference though. The main thing is that the block results are not in order)

This would be really nice.